src/Storefront/Resources/app/storefront/src/component-system/README.md

```diff
@@ -0,0 +1,361 @@
+    onContentUpdate(mutationRecord) {}
+
+    // Handle attribute changes
+    onAttributeUpdate(mutationRecord) {}
+
+    // Custom methods
+    setupEventListeners() {
+        this.el.addEventListener('click', this.handleClick.bind(this));
+    }
+
+    handleClick(event) {
+        // Custom logic
+    }
+}
+
+window.Shopware.registerComponent('my-component', MyComponent);
+```
+
+### Component Registration
+
+Components are registered with the global Shopware instance. You can do this simply in the same file as your component.
+
+```javascript
+// Register the component
+window.Shopware.registerComponent('my-component', MyComponent);
+```
+
+### Automatic Initialization
+
+To define which elements the component should be applied to, you can define a static property `selector` in your component class. The component will then automatically be initialized on elements that match the selector. This also applies to elements that might be added later to the document. You do not need to do this manually.
+
+```javascript
+class MyComponent extends ShopwareComponent {
+
+    static selector = '[data-my-component]';
+    
+    init() {
+        console.log('Component automatically initialized!');
+    }
+}
+```
+
+## Component Configuration
+
+### Data Attributes
+
+Components can be configured through data attributes using the pattern `data-{component-name}-options`. For example, you can pass information form Twig into your component. The options should be passed as a JSON string.
+
+```Twig
+
+{% set componentOptions = {
+    foo: "bar" 
+    test: true
+} %}
+
+<div data-my-component 
+     data-my-component-options="{{ componentOptions|json_encode }}">
+</div>
+```
+
+The component name is automatically converted to dash-case for the data attribute. The passed options are merged with the default options that you define as a static property in your component class.
+
+## Directory Structure & Component Script Loading
+
+Component scripts are automatically loaded when the corresponding Twig component is used within the page. Your component script should have the same name as the Twig file of your component and should be placed in the same directory. Shopware will automatically collect all component script files and include them into the page if the component is used on a specific page.
+
+Example structure:
+
+```
+views/
+  components/
+    MyComponentNamespace/
+      MyComponent.thml.twig
+      MyComopnent.js
+      MyComponent.scss
+```
+
+## Twig Component Integration
+
+To integrate the script with your corresponding Twig component you have to ensure that the desired element within your component template has the necessary data attributes.
+
+You can build the options individually from your Twig component properties and other data, or simply pass through the component properties to your JavaScript component.
+
+```twig
+{# views/component/MyComponent.html.twig #}
+
+{% props
+    foo = "bar",
+    custom = true,
```

Comment:
Should we pass all props as an example? Or should we have an extra prop for the JS plugin options that then gets passed. Maybe not all of the twig props are automatically needed in the JS.

Reply:
I added the `jsOptions` example as well, but I found it a nice example for a "real" component-like feeling. I think sometimes this might be very useful, especially for smaller components, so the JS and the Twig component work perfectly in conjunction.